.github/workflows/teamcity-services-diff-check.yml

```diff
@@ -10,6 +10,15 @@ on:
       - 'mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt'
       - 'mmv1/third_party/terraform/.teamcity/components/inputs/services_beta.kt'
 jobs:
+  check-pr:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check for New Services
+        run: |
+          newServices=$(($(git diff --name-only --diff-filter=A origin/main HEAD | grep "mmv1/products" | wc -l)))
+          if [ "$newServices" != "0" ];then
+          exit 1
```

Comment:
Lets pair on this section, but here's a list of things that we should discuss when we pair:

- What is the if condition checking here? Is it correct?
- Adding feedback to the user about the significance of this step, e.g. via an `echo` command
- Ways to make the newServices count more accurate using patterns/regex with `grep`
   -  On Mac, `grep -e "mmv1/products/.*/product.yaml"` would help identify when new services are added
- Use of [outputs](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs) and [conditional job execution](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution) to control when the GHA runs (replacing `exit 1`)